runtime/appruntime/apisdk/service/service.go

```diff
@@ -120,6 +126,21 @@ func (mgr *Manager) RegisterService(i Initializer) {
 	mgr.svcInit = append(mgr.svcInit, i)
 }
 
+func (mgr *Manager) IsHosting(i Initializer) bool {
```

Comment:
Maybe clearer to have this take a `name string` argument? And then in `RegisterService` move the `!mgr.IsHosting(...)` if statement below `name := i.ServiceName()`